docs/deployment_kubernetes_docker.md

```diff
@@ -0,0 +1,268 @@
+from bentoml.artifact import FastaiModelArtifact
+from bentoml.handlers import FastaiImageHandler
+
+@artifacts([FastaiModelArtifact('pet_classifier')])
+@env((auto_pip_dependencies=True)
+class PetClassification(BentoService):
+
+    @api(FastaiImageHandler)
+    def predict(self, image):
+        result = self.artifacts.pet_classifier.predict(image)
+        return str(result)
+```
+
+This code defines a model server using `Fastai` model, asks BentoML to figure out
+the required PyPi packages automatically. It also defines an API called `predict`, that
+is the entry point to access this model server. The API expects a `Fastai`
+`ImageData` object as its input data.
+
+Run the following code to create a BentoService SavedBundle with the pet classification
+model. A BentoService SavedBundle is a versioned file archive ready for production
+deployment. The archive contains the model service defined above, python code
+dependencies and PyPi dependencies, and the trained pet classification model:
+
+```python
+# Create a PetClassification instance
+from pet_classification import PetClassification
+service = PetClassification()
+
+#  Pack the newly trained model artifact
+service.pack('pet_classifier', learn)
+
+# Save the prediction service to disk for model serving
+service.save()
+```
+
+BentoML automatically process the incoming data into required data format defined in the
+API. For the pet classifier BentoService defined above, incoming data will transform to
+fastai `ImageData` object.
+
+Use BentoML CLI to start a local API model server:
+
+```bash
+bentoml serve PetClassification:latest
+```
+
+Make `curl` request in another terminal to get the prediction result:
+
+```bash
+# Replace PATH_TO_TEST_IMAGE_FILE with one of the image from {path_img}
+# An example path: /Users/user_name/.fastai/data/oxford-iiit-pet/images/shiba_inu_122.jpg
+curl -i \
+    --request POST \
+    --header "Content-Type: multipart/form-data" \
+    -F "image=@PATH_TO_TEST_IMAGE_FILE" \
+    localhost:5000/predict
+```
+
+## Deploy model server to Kubernetes
+
+### Build model server image
+
+Find the file directory of the SavedBundle with `bentoml get` command, which is
+directory structured as a docker build context. Running docker build with this
+directory produces a docker image containing the model API server. Replace
+docker_username with your Docker Hub username and run the following code:
+
+```bash
+saved_path=$(bentoml get PetClassifier:latest -q | jq -r ".uri.uri")
+
+docker build -t {docker_username}/pet-classifier .
+
+docker push {docker_username}/pet-classifier
+```
+
+Use `docker run` command to test the docker image locally:
+
+```bash
+docker run -p 5000:5000 {docker_username}/pet-classifier
+```
+
+In another terminal window, use the `curl` command from above to get the prediction result.
+
+### Deploy to Kubernetes
+
+The following code creates a kubernetes resources spec. Replace `{docker_username}`
```

Comment:
The following is an example YAML file for specifying the resources required to run and expose a BentoML model server in a Kubernetes cluster. Replace `{docker_username}` with your Docker Hub username and save it to `pet-classifier.yaml` file: